solidity/contracts/bridge/Bridge.sol

```diff
@@ -1339,6 +1406,155 @@ contract Bridge is Ownable, EcdsaWalletOwner {
+    ///      - Wallet can be challenged for the given signature only once.
+    function submitFraudChallenge(
+        bytes memory walletPublicKey,
+        bytes32 sighash,
+        Frauds.RSVSignature calldata signature
+    ) external payable {
+        bytes memory compressedWalletPublicKey = EcdsaLib.compressPublicKey(
+            walletPublicKey.slice32(0),
+            walletPublicKey.slice32(32)
+        );
+
+        bytes20 walletPubKeyHash = bytes20(compressedWalletPublicKey.hash160());
+
+        Wallets.Wallet storage wallet = wallets.registeredWallets[
+            walletPubKeyHash
+        ];
+
+        require(
+            wallet.state == Wallets.WalletState.Live ||
+                wallet.state == Wallets.WalletState.MovingFunds,
+            "Wallet is neither in Live nor MovingFunds state"
+        );
+
+        frauds.submitFraudChallenge(walletPublicKey, sighash, signature);
+    }
+
+    /// @notice Allows to defeat a pending fraud challenge against a wallet if
+    ///         the transaction that spends the UTXO follows the protocol rules.
+    ///         In order to defeat the challenge the same `walletPublicKey` and
+    ///         signature (represented by `r`, `s` and `v`) must be provided as
+    ///         were used in the fraud challenge. Additionally a preimage must
+    ///         be provided which was used to calculate the sighash during input
+    ///         signing. The fraud challenge defeat attempt will only succeed if
+    ///         the inputs in the preimage are considered honestly spent by the
+    ///         wallet. Therefore the transaction spending the UTXO must be
+    ///         proven in the Bridge before a challenge defeat is called.
+    ///         If successfully defeated, the fraud challenge is closed and the
+    ///         amount of ether deposited by the challenger is sent to the
+    ///         treasury.
+    /// @param walletPublicKey The public key of the wallet in the uncompressed
+    ///        and unprefixed format (64 bytes).
+    /// @param preimage The preimage which produces sighash used to generate the
+    ///        ECDSA signature that is the subject of the fraud claim. It is a
+    ///        serialized subset of the transaction. The exact subset used as
+    ///        the preimage depends on the transaction input the signature is
+    ///        produced for. See BIP-143 for reference.
+    /// @param signature Bitcoin signature in the R/S/V format.
+    /// @param witness Flag indicating whether the preimage was produced for a
+    ///        witness input. True for witness, false for non-witness input.
+    /// @dev Requirements:
+    ///      - `walletPublicKey`, signature (represented by `r`, `s` and `v`) as
+    ///        well `sighash` calculated as hash256(preimage) must identify an
+    ///        open fraud challenge
+    ///      - the preimage must be generated according to the protocol
```

Comment:
I wonder if this is clear enough. One could say generating a preimage of any Bitcoin transaction according to the Bitcoin protocol fulfills this requirement. Instead, we could say something like: *the preimage must be a valid preimage of a transaction generated according to the protocol rules and already proved in the Bridge*.

Reply:
Done in 9c85e4483857e9487c8d7bddcaf9bd2df792dfdd.